packages/devtools_app/integration_test/test/live_connection/eval_and_browse_test.dart

```diff
@@ -116,5 +124,77 @@ class _EvalTester {
     await tester.pumpAndSettle();
     await simulateKeyDownEvent(LogicalKeyboardKey.enter);
     await simulateKeyUpEvent(LogicalKeyboardKey.enter);
+    await tester.pump(longPumpDuration);
+  }
+
+  Future<void> switchToSnapshotsAndTakeOne() async {
+    // Open memory screen.
+    await switchToScreen(tester, ScreenMetaData.memory);
+
+    // Close warning and chart to get screen.
+    await tapAndPump(find.byKey(DebugModeMemoryMessage.closeKey));
+    await tapAndPump(find.text(PrimaryControls.memoryChartText));
+
+    // Increase cinsole.
+    await tester.drag(
+      find.byKey(ConsolePaneHeader.theKey),
+      const Offset(0, -320),
+    );
+    await tester.pumpAndSettle();
+
+    // Switch to diff tab.
+    await tapAndPump(find.text('Diff Snapshots'));
+
+    logStatus('Started taking snapshot.');
+    // Take snapshot.
+    const snapshotDuration = Duration(seconds: 20);
+    await tapAndPump(
+      find.byIcon(iconToTakeSnapshot),
+      duration: snapshotDuration,
+    );
+    logStatus('Finished taking snapshot.');
+
+    // Sort by class.
+    await tapAndPump(find.text('Class'));
+
+    // Select class.
+    await tapAndPump(find.text('MyApp'));
+  }
+
+  /// Taps and settles.
+  ///
+  /// If [next] is provided, will repeat the tap till [next] combined with [at] returns results.
+  /// If [next] is not null returns [next]  combined with [at, otherwise returns [finder].
+  Future<Widget?> tapAndPump(
+    Finder finder, {
+    Duration? duration,
+    Finder? next,
+  }) async {
+    int tryNumber = 0;
+
+    Future<void> action() async {
+      logStatus('tapping #$tryNumber to find \n'
+          '[$finder]\n');
+      tryNumber++;
```

Comment:
Nit: let's update `action` to take `action(int attempt)` instead of incrementing the try counter here. That way we can declare `int tryNumber` just before the loop and keep the increment logic in the loop.